src/main/scala/chisel3/util/experimental/decode/TruthTable.scala

```diff
@@ -4,7 +4,7 @@ package chisel3.util.experimental.decode
 
 import chisel3.util.BitPat
 
-final class TruthTable(val table: Map[BitPat, BitPat], val default: BitPat) {
+sealed class TruthTable(val table: Seq[(BitPat, BitPat)], val default: BitPat) {
```

Comment:
```suggestion
sealed class TruthTable private (val table: Seq[(BitPat, BitPat)], val default: BitPat) {
```
So that users need to construct this via the companion object apply method.